src/App.tsx

```diff
@@ -1,61 +1,138 @@
+import React, { useEffect, useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { Todo, Error } from './types';
+
+const getUserById = (userId: number) => {
+  return usersFromServer.find(user => user.id === userId) || null;
+};
+
+const todosWithUsers = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [todos, setTodos] = useState(todosWithUsers);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [selectedUserId, setSelectedUserId] = useState(0);
+
+  const [showTitleError, setShowTitleError] = useState(false);
+  const [showUserError, setShowUserError] = useState(false);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const validTitle = event.target.value.replace(/[^a-zA-Za-яA-Я ]/g, '');
+
+    setNewTodoTitle(validTitle);
+  };
+
+  const addNewTodo = (title: string, userId: number, id: number) => {
+    const newTodo: Todo = {
+      id: id + 1,
+      title,
+      completed: false,
+      userId,
+      user: getUserById(userId),
+    };
+
+    setTodos([...todos, newTodo]);
+  };
+
+  const currentMaxId = Math.max(...todos.map(todo => todo.id));
+
+  const handleFormSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    setShowTitleError(!newTodoTitle);
+    setShowUserError(!selectedUserId);
+
+    if (newTodoTitle && selectedUserId) {
```

Comment:
Consider using a more descriptive variable name instead of `id`. For example, `maxTodoId`. 🤔